From 37c8ca7031c2a70111b39c6d2b545946ee5dbb9d Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Tue, 30 Aug 2016 18:07:36 +0500
Subject: [PATCH] obey CMAKE_INSTALL_LIBDIR

The CMAKE_INSTALL_* directories provide a mechanism to
relocate installations so rather than doing this through
hardcoded variable they should be used whereever possible.
This fixes installation to required directory.

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 layers/CMakeLists.txt   | 4 ++--
 layersvt/CMakeLists.txt | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/layers/CMakeLists.txt b/layers/CMakeLists.txt
index 076b847..cfab1bf 100644
--- a/layers/CMakeLists.txt
+++ b/layers/CMakeLists.txt
@@ -81,7 +81,7 @@ else()
     target_link_Libraries(VkLayer_${target} VkLayer_utils)
     add_dependencies(VkLayer_${target} generate_vk_layer_helpers)
     set_target_properties(VkLayer_${target} PROPERTIES LINK_FLAGS "-Wl,-Bsymbolic")
-    install(TARGETS VkLayer_${target} DESTINATION ${PROJECT_BINARY_DIR}/install_staging)
+    install(TARGETS VkLayer_${target} DESTINATION ${CMAKE_INSTALL_LIBDIR})
     endmacro()
 endif()
 
@@ -148,7 +148,7 @@ if (WIN32)
     add_library(VkLayer_utils STATIC vk_layer_config.cpp vk_layer_extension_utils.cpp vk_layer_utils.cpp)
 else()
     add_library(VkLayer_utils SHARED vk_layer_config.cpp vk_layer_extension_utils.cpp vk_layer_utils.cpp)
-    install(TARGETS VkLayer_utils DESTINATION ${PROJECT_BINARY_DIR}/install_staging)
+    install(TARGETS VkLayer_utils DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
 
 add_vk_layer(core_validation core_validation.cpp vk_layer_table.cpp vk_safe_struct.cpp descriptor_sets.cpp)
diff --git a/layersvt/CMakeLists.txt b/layersvt/CMakeLists.txt
index 8595418..ae05d07 100644
--- a/layersvt/CMakeLists.txt
+++ b/layersvt/CMakeLists.txt
@@ -84,7 +84,7 @@ else()
     target_link_Libraries(VkLayer_${target} VkLayer_utilsvt)
     add_dependencies(VkLayer_${target} generate_vt_helpers)
     set_target_properties(VkLayer_${target} PROPERTIES LINK_FLAGS "-Wl,-Bsymbolic")
-    install(TARGETS VkLayer_${target} DESTINATION ${PROJECT_BINARY_DIR}/install_staging)
+    install(TARGETS VkLayer_${target} DESTINATION ${CMAKE_INSTALL_LIBDIR})
     endmacro()
 endif()
 
@@ -166,7 +166,7 @@ if (WIN32)
     add_library(VkLayer_utilsvt STATIC ../layers/vk_layer_config.cpp ../layers/vk_layer_extension_utils.cpp ../layers/vk_layer_utils.cpp)
 else()
     add_library(VkLayer_utilsvt SHARED ../layers/vk_layer_config.cpp ../layers/vk_layer_extension_utils.cpp ../layers/vk_layer_utils.cpp)
-    install(TARGETS VkLayer_utilsvt DESTINATION ${PROJECT_BINARY_DIR}/install_staging)
+    install(TARGETS VkLayer_utilsvt DESTINATION ${CMAKE_INSTALL_LIBDIR})
 endif()
 
 # VulkanTools layers
-- 
1.9.1

